File: java/nanocontainer/src/test-comp/TestComp2.java

public class TestComp2 {

	public TestComp2(TestComp tc, StringBuffer sb) {
        sb.append("-TestComp2");
        System.out.println("--> !!!");
	    if (tc == null) {
			throw new NullPointerException();
		}
	}
}

File: trunk/dotnet/nanocontainer/src/TestComp2/TestComp2.cs
using System;
using System.Text;
using PicoContainer;

public class TestComp2 : IStartable 
{
	public TestComp2(TestComp tc, StringBuilder sb) 
	{
		sb.Append("-TestComp2");
		if (tc == null) 
		{
			throw new NullReferenceException();
		}
	}

	public void Start() {}
	public void Stop() {}
}
